app/components/publish-actions.tsx

```diff
@@ -126,8 +126,7 @@ const Embed = ({
         iframeHeight={iframeHeight}
         configKey={configKey}
         locale={locale}
-        state={state}
-        isSingleChart={state?.chartConfigs?.length === 1}
+        shouldAllowDisablingBorder={state?.chartConfigs?.length === 1}
```

Comment:
@bprusinowski not quite sure if I get it both seem to be using `state?.chartConfigs?.length === 1` and 

```javascript
const isSingleChart = config.data.chartConfigs.length === 1;

Reply:
Ah, here I mean that we should have a shared function, like `const shouldAllowDisablingBorder = () => ...` that you'd use both here and in the /v/ page – now the condition is the same, but it's actually defined in two separate places, which can cause issues in case we need to change this (already looks like we had the condition in one place, but not the other).

Also see the comment that explains why we should add the `tab` check as was done previously: https://github.com/visualize-admin/visualization-tool/pull/1826#discussion_r1837713283 (I think in case of single-chart dashboard, we don't want to have the UI either) 👀

Reply:
Or is it the only place that affects both components? If so, it would be only about the `tab` check addition 👍

Reply:
I see, i see so you mean something like

```javascript 
state?.chartConfigs?.length === 1 && state?.layout.type !== "dashboard"
```

Reply:
I think I would use the `=== "tab"`, we also don't want this option for `singleURLs` layout :) I guess somewhere near the Embed component, could be in the same file 👍